Python/optimizer_bytecodes.c

```diff
@@ -370,6 +370,20 @@ dummy_func(void) {
         res = sym_new_type(ctx, &PyUnicode_Type);
     }
 
+    op(_BINARY_OP_SUBSCR_TUPLE_INT, (left, right -- res)) {
+        assert(sym_matches_type(left, &PyTuple_Type));
+        if (sym_is_const(ctx, right)) {
+            assert(PyLong_CheckExact(sym_get_const(ctx, right)));
+            long index = PyLong_AsLong(sym_get_const(ctx, right));
+            assert(index >= 0);
+            assert(index < sym_tuple_length(left));
+            res = sym_tuple_getitem(ctx, left, index);
```

Comment:
This is a bit unfortunate. I would wish we could automatically evaluate this, but it seems tuples are a special category so we can't.

https://github.com/python/cpython/pull/132733

Reply:
I haven't fully digested your PR yet, but maybe there is some way to extend it to support tuples as well?